src/node_perf.cc

```diff
@@ -21,13 +25,36 @@ using v8::Object;
 using v8::String;
 using v8::Value;
 
+// Microseconds in a second, as a float.
+#define MICROS_PER_SEC 1e6
+// Microseconds in a millisecond, as a float.
+#define MICROS_PER_MILLIS 1e3
+
+// https://w3c.github.io/hr-time/#dfn-time-origin
 const uint64_t timeOrigin = PERFORMANCE_NOW();
+// https://w3c.github.io/hr-time/#dfn-time-origin-timestamp
+const double timeOriginTimestamp = GetCurrentTimeInMicroseconds();
 uint64_t performance_node_start;
 uint64_t performance_v8_start;
 
 uint64_t performance_last_gc_start_mark_ = 0;
 v8::GCType performance_last_gc_type_ = v8::GCType::kGCTypeAll;
 
+double GetCurrentTimeInMicroseconds() {
+#ifdef _WIN32
+  FILETIME ft;
+  GetSystemTimeAsFileTime(&ft);
+  uint64_t filetime_int = static_cast<uint64_t>(ft.dwHighDateTime) << 32 |
+                          ft.dwLowDateTime;
+  // FILETIME is measured in terms of 100 ns. Convert that to 1 us (1000 ns).
+  return filetime_int / 10.;
+#else
+  struct timeval tp;
+  gettimeofday(&tp, nullptr);
+  return MICROS_PER_SEC * tp.tv_sec + tp.tv_usec;
+#endif
+}
```

Comment:
Is it guaranteed that the result of this call is compatible with `uv_hrtime` that we use for `PERFORMANCE_NOW`? I guess the test ensures that, but I feel like I’m missing something here…

Reply:
This is not compatible with `ur_hrtime()`, nor is it intended to be compatible. This function returns the real Unix time at which the function is called, while `uv_hrtime()` returns a monotonic timestamp based on some arbitrary time in the past.

The only place this is used is in the newly created `timeOriginTimestamp` global variable, to which `performance.timeOrigin` is set. `performance.timeOrigin` is intended to be a Unix timestamp (see #17893).